torch/package/package_importer.py

```diff
@@ -296,6 +299,16 @@ def _read_extern(self):
             .splitlines(keepends=False)
         )
 
+    def _read_selective_extern(self):
+        try:
+            return (
+                self.zip_reader.get_record(".data/selective_extern_packages")
+                .decode("utf-8")
+                .splitlines(keepends=False)
+            )
+        except BaseException:
```

Comment:
Catching BaseException is too general here. We should just be checking whether file exists and bailing if it doesn't.